src/Analyzers/CSharp/Tests/PopulateSwitch/PopulateSwitchExpressionTests.cs

```diff
@@ -1367,5 +1368,41 @@ void M()
 }
 ");
         }
+
+        [Fact, WorkItem(61594, "https://github.com/dotnet/roslyn/issues/61594")]
+        public async Task DoNotKeepNullability()
+        {
+            await TestInRegularAndScript1Async(
+@"#nullable enable
+
+static class MyEnumExtensions
+{
+    public static string ToAnotherEnum(this MyEnum? myEnumValue)
+        => myEnumValue [||]switch
+        {
+        };
+}
+
+enum MyEnum
+{
+    Value1, Value2
+}",
+@"#nullable enable
+
+static class MyEnumExtensions
+{
+    public static string ToAnotherEnum(this MyEnum? myEnumValue)
+        => myEnumValue switch
+        {
+            MyEnum.Value1 => throw new System.NotImplementedException(),
+            MyEnum.Value2 => throw new System.NotImplementedException(),
+        };
```

Comment:
Why null pattern isn't produced?

Reply:
Don't understand what exactly you mean. Can you give a piece of code?

Reply:
I was expecting the fixed code to be:

```csharp
#nullable enable
static class MyEnumExtensions
{
    public static string ToAnotherEnum(this MyEnum? myEnumValue)
        => myEnumValue switch
        {
            MyEnum.Value1 => throw new System.NotImplementedException(),
            MyEnum.Value2 => throw new System.NotImplementedException(),
            null => throw new System.NotImplementedException(),
        };
}

enum MyEnum
{
    Value1, Value2
}"
```

Reply:
i'm fine either way.  up to you @DoctorKrolic if you want to enhance in that way.

Reply:
The original issue is about wrong `?.` used, but I think it is preferable to implement Youssef1313's suggestion here as well

Reply:
@DoctorKrolic Cool.   I'll wait on you for that.  Feel free to ping me when ready.